### PR TITLE
Update core-contracts to latest cadence 1.0 version

### PR DESCRIPTION
**Pull request by joshuahannan**
Updates to the latest cadence 1.0 version of the token standards and core contracts

This is needed to a new version of the cadence 1.0 CLI and emulator release

### CODE REVIEW

**Comment by joshuahannan** [comment]
I'm not sure why the CI is failing, the results don't load for me. Anyone know what we need to do?

**Comment by jordanschalm** [comment]
> I'm not sure why the CI is failing, the results don't load for me. Anyone know what we need to do?

It looks like the execution tests with constants for expected state commitments:
- Test_ExecutingSystemCollection
- TestBootstrapLedger
- TestExecutionFlow

You'll need to change the state commitment constant in the code to the new state commitment, after the smart contract updated, which will be printed in the error message from the test. Here's an example PR where I did this recently: https://github.com/onflow/flow-go/pull/5027. Changes are in:
- engine/execution/state/bootstrap/bootstrap_test.go
- utils/unittest/execution_state.go

**Comment by joshuahannan** [comment]
anyone know what this error means?

```
panic: 

mock: Unexpected Method Call
-----------------------------

ExecutionTransactionExecuted(time.Duration,int,uint64,uint64,int,int,bool)
		0: 24058000
		1: 1
		2: 0x102
		3: 0xd64dd2
		4: 3
		5: 1465
		6: false

The closest call I have is: 

ExecutionTransactionExecuted(string,string,string,string,int,int,bool)
		0: "mock.Anything"
		1: "mock.Anything"
		2: "mock.Anything"
		3: "mock.Anything"
		4: 3
		5: 1421
		6: false


Diff: 0: PASS:  (time.Duration=24.058ms) == (string=mock.Anything)
	1: PASS:  (int=1) == (string=mock.Anything)
	2: PASS:  (uint64=258) == (string=mock.Anything)
	3: PASS:  (uint64=14044626) == (string=mock.Anything)
	4: PASS:  (int=3) == (int=3)
	5: FAIL:  (int=1465) != (int=1421)
	6: PASS:  (bool=false) == (bool=false)

goroutine 239 [running]:
github.com/stretchr/testify/mock.(*Mock).fail(0x14000b2fa40, {0x101cdf659?, 0x140008b5ce0?}, {0x1400169c200?, 0x7?, 0x7?})
	/opt/homebrew/opt/go@1.20/bin/pkg/mod/github.com/stretchr/testify@v1.8.4/mock/mock.go:332 +0x15c
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0x14000b2fa40, {0x10255ab9e, 0x1c}, {0x140003a6000, 0x7, 0x7})
	/opt/homebrew/opt/go@1.20/bin/pkg/mod/github.com/stretchr/testify@v1.8.4/mock/mock.go:492 +0x500
github.com/stretchr/testify/mock.(*Mock).Called(0x5b9?, {0x140003a6000, 0x7, 0x7})
	/opt/homebrew/opt/go@1.20/bin/pkg/mod/github.com/stretchr/testify@v1.8.4/mock/mock.go:464 +0x114
github.com/onflow/flow-go/module/mock.(*ExecutionMetrics).ExecutionTransactionExecuted(0x14000b2fa40, 0x140002a6201?, 0x0?, 0x0?, 0x0?, 0x140006d6200?, 0x124?, 0x0)
	/Users/joshuahannan/dapperlabs/flow/flow-go/module/mock/execution_metrics.go:101 +0x22c
github.com/onflow/flow-go/engine/execution/computation/computer.(*resultCollector).processTransactionResult(_, {{{0xd, 0x33, 0x97, 0x74, 0x3d, 0x1, 0x3d, 0xbf, 0xe3, ...}, ...}, ...}, ...)
	/Users/joshuahannan/dapperlabs/flow/flow-go/engine/execution/computation/computer/result_collector.go:254 +0x8fc
github.com/onflow/flow-go/engine/execution/computation/computer.(*resultCollector).runResultProcessor(0x140025854a0)
	/Users/joshuahannan/dapperlabs/flow/flow-go/engine/execution/computation/computer/result_collector.go:328 +0x1d8
created by github.com/onflow/flow-go/engine/execution/computation/computer.newResultCollector
	/Users/joshuahannan/dapperlabs/flow/flow-go/engine/execution/computation/computer/result_collector.go:127 +0x7cc
FAIL	github.com/onflow/flow-go/engine/execution/computation/computer	0.814s
```

**Comment by jordanschalm** [comment]
> anyone know what this error means?

@joshuahannan  It's expecting that mock function to be called with particular arguments. But the `eventSize` argument is different:
```
	5: FAIL:  (int=1465) != (int=1421)
```

I pushed this commit to your branch which should fix that test: https://github.com/onflow/flow-go/pull/5064/commits/5d7a937d619ffc9aed68f67888967ab1a371aabc

**Comment by joshuahannan** [comment]
okay, everything is passing locally for me when I run `make test`

**Review by janezpodhostnik — APPROVED** [review]

**Review by jordanschalm — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

**Comment by joshuahannan** [comment]
How do I run all the other CI tests?

**Comment by jordanschalm** [comment]
> How do I run all the other CI tests?

I think these are not running because the feature branch hasn't been updated to include some CI changes @gomisha did recently. Last change on the [feature branch CI config](https://github.com/onflow/flow-go/blob/feature/stable-cadence/.github/workflows/ci.yml) is 3 months ago, last change on [master](https://github.com/onflow/flow-go/blob/master/.github/workflows/ci.yml) is 3 weeks ago.

I don't completely understand how GH Actions decides which `ci.yml`(s) to use, but it seems like here it is using both the CI jobs from your branch (same as the feature branch) and from `master` (because it is the default branch (?) ). My assumption is syncing the feature branch with master, then merging this PR would unblock you. Misha is also back on Monday, and can help here.

**Comment by turbolent** [comment]
Updated the feature branch in #5102. Once merged, and this PR has in turn merged in/rebased on the feature branch, CI should work again

**Review by turbolent — APPROVED** [review]

**Pull request merged**
